src/Transform/ONNX/ConstProp.cpp

```diff
@@ -217,6 +218,33 @@ struct ElementWiseBinaryOpImpl<ONNXMaxOp, T> {
   static T eval(T lhs, T rhs) { return std::max<T>(lhs, rhs); }
 };
 
+template <typename T>
+struct ElementWiseBinaryOpImpl<ONNXModOp, T, EnableNotBool<T>> {
+  static T eval(T lhs, T rhs) {
+    // This is the original calculation for mod
+    int mod = lhs - floor(lhs / rhs) * rhs;
+
+    // Check if the values are floats
+    if (lhs != floor(lhs) && rhs != floor(rhs)) {
```

Comment:
Since this is a struct template on type T, it is better to have two instances:
```
template <>
struct ElementWiseBinaryOpImpl<ONNXModOp, int64, EnableNotBool<int64>>
```

and 

```
template <>
struct ElementWiseBinaryOpImpl<ONNXModOp, double, EnableNotBool<double>>
```

Reply:
I updated the code to add two struct templates with the int and double data types